app/plugin.go

```diff
@@ -465,55 +557,96 @@ func (a *App) GetMarketplacePlugins(filter *model.MarketplacePluginFilter) ([]*m
 		ServerVersion: model.CurrentVersion,
 	})
 	if err != nil {
-		return nil, model.NewAppError("GetMarketplacePlugins", "app.plugin.marketplace_plugins.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, model.NewAppError("getRemotePlugins", "app.plugin.marketplace_client.failed_to_fetch", nil, err.Error(), http.StatusInternalServerError)
 	}
 
 	for _, p := range marketplacePlugins {
-		if p.Manifest == nil || !pluginMatchesFilter(p.Manifest, filter.Filter) {
+		if p.Manifest == nil {
+			continue
+		}
+
+		result[p.Manifest.Id] = &model.MarketplacePlugin{BaseMarketplacePlugin: p}
+	}
+
+	return result, nil
+}
+
+// mergePrepackagePlugins merges pre-packaged plugins to remote marketplace plugins list.
+func (a *App) mergePrepackagePlugins(remoteMarketplacePlugins map[string]*model.MarketplacePlugin) *model.AppError {
```

Comment:
Let's add some tests for mergePrepackagePlugins and mergeLocalPlugins ? I didn't see them.

Reply:
They were tested as a whole with all the test cases of GetMarketplacePlugins in `api4/plugin_test`:
https://github.com/mattermost/mattermost-server/blob/855c0bdaeed0474666c15787934b97d01a936d34/api4/plugin_test.go#L789
https://github.com/mattermost/mattermost-server/blob/855c0bdaeed0474666c15787934b97d01a936d34/api4/plugin_test.go#L933
https://github.com/mattermost/mattermost-server/blob/855c0bdaeed0474666c15787934b97d01a936d34/api4/plugin_test.go#L462
https://github.com/mattermost/mattermost-server/blob/855c0bdaeed0474666c15787934b97d01a936d34/api4/plugin_test.go#L789